numpyro/contrib/tfp/distributions.py

```diff
@@ -0,0 +1,182 @@
+    with NumPyro internal.
+    """
+    def __call__(self, *args, **kwargs):
+        key = kwargs.pop('rng_key')
+        return self.sample(*args, seed=key, **kwargs)
+
+    @property
+    def support(self):
+        bijector = self._default_event_space_bijector()
+        if bijector is not None:
+            return BijectorConstraint(bijector)
+        else:
+            return None
+
+    @property
+    def is_discrete(self):
+        # XXX: this should cover most cases
+        return self.support is None
+
+
+class InverseGamma(tfd.InverseGamma):
+    arg_constraints = {"concentration": constraints.positive, "scale": constraints.positive}
+
+
+class OneHotCategorical(tfd.OneHotCategorical):
+    arg_constraints = {"logits": constraints.real_vector}
+    has_enumerate_support = True
+    support = constraints.simplex
+    is_discrete = True
+
+    def enumerate_support(self, expand=True):
+        n = self.event_shape[-1]
+        values = jnp.identity(n, dtype=canonicalize_dtype(self.dtype))
+        values = values.reshape((n,) + (1,) * len(self.batch_shape) + (n,))
+        if expand:
+            values = jnp.broadcast_to(values, (n,) + self.batch_shape + (n,))
+        return values
+
+
+class OrderedLogistic(tfd.OrderedLogistic):
+    arg_constraints = {"cutpoints": constraints.ordered_vector, "loc": constraints.real}
+
+
+class Pareto(tfd.Pareto):
+    arg_constraints = {"concentration": constraints.positive, "scale": constraints.positive}
+
+
+__all__ = ['BijectorConstraint', 'BijectorTransform', 'TFPDistributionMixin']
+_len_all = len(__all__)
+for _name, _Dist in tfd.__dict__.items():
+    if not isinstance(_Dist, type):
+        continue
+    if not issubclass(_Dist, tfd.Distribution):
+        continue
+    if _Dist is tfd.Distribution:
+        continue
+
+    try:
+        _PyroDist = locals()[_name]
+    except KeyError:
+        _PyroDist = type(_name, (_Dist, TFPDistributionMixin), {})
+        _PyroDist.__module__ = __name__
+        if hasattr(numpyro_dist, _name):
+            numpyro_dist_class = getattr(numpyro_dist, _name)
+            # resolve FooProbs/FooLogits namespaces
+            if type(numpyro_dist_class).__name__ == "function":
+                if not hasattr(numpyro_dist, _name + "Logits"):
+                    continue
+                numpyro_dist_class = getattr(numpyro_dist, _name + "Logits")
+            _PyroDist.arg_constraints = numpyro_dist_class.arg_constraints
+            _PyroDist.has_enumerate_support = numpyro_dist_class.has_enumerate_support
+            _PyroDist.enumerate_support = numpyro_dist_class.enumerate_support
+        locals()[_name] = _PyroDist
+
+    _PyroDist.__doc__ = '''
+    Wraps `{}.{} <https://www.tensorflow.org/probability/api_docs/python/tfp/substrates/jax/distributions/{}>`_
+    with :class:`~numpyro.contrib.tfp.distributions.TFPDistributionMixin`.
+    '''.format(_Dist.__module__, _Dist.__name__, _Dist.__name__)
```

Comment:
I added `_Dist.__doc__` but it does not render well because it is written in Markdown format. Currently `help(dist)` already has a lot of information.